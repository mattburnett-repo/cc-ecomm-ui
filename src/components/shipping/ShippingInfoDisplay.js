
import CartSummary from '../../widgets/CartSummary'

// TODO: should have PropTypes in this, and other, didsplay component/s
// import PropTypes from 'prop-types';

export default function ShippingInfoDisplay (props) {
    const { handleGoToPaymentInfo } = props.handlers
    const { savedAddresses } = props

    return (
        <div role="presentation" aria-label="shipping-info">
            <h3>Shipping Info</h3>
            <CartSummary />
            {/* <form onSubmit={handleSaveShippingInfo} method="post"> */}
            <form onSubmit={handleGoToPaymentInfo} method="post">  
                {(savedAddresses.length > 0) ?
                    (<div role='presentation' aria-label="saved-addresses-display">
                        {/* TODO: this should be a dropdown, and should auto-fill the address fields after onChange event */}
                        You have {savedAddresses.length} saved address/es
                    </div>)
                    : <></>
                }
                <div>
                    <label htmlFor="first-name">First Name:</label>
                    <input id='first-name' name="firstName" aria-label="first-name"/>
                </div>
                <div>
                    <label htmlFor="last-name">Last Name:</label>
                    <input id='last-name' name="lastName" aria-label="last-name"/>
                </div>
                <div>
                    <label htmlFor="address-1">Address 1:</label>
                    <input id='address-1' name="address1" aria-label="address-1"/>
                </div>
                <div>
                    <label htmlFor="address-2">Address 2:</label>
                    <input id='address-2' name="address2" aria-label="address-2"/>
                </div>
                <div>
                    <label htmlFor="city">City:</label>
                    <input id='city' name="city" aria-label="city"/>
                </div>
                <div>
                    <label htmlFor="state-province">State / Province:</label>
                    <input id='state-province' name="stateProvince" aria-label="state-province"/>
                </div>
                <div>
                    <label htmlFor="postal-code">Postal Code:</label>
                    <input id='postal-code' name="postalCode" aria-label="postal-code"/>
                </div>
                <div>
                    <label htmlFor="country">Country:</label>
                    <input id='country' name="country" aria-label="country"/>
                </div>

                {/* TODO: good idea here, but not in wireframe. Come back and work on it later */}
                {/* <button type="submit" aria-label="save-shipping-info">Save Shipping Info For Later Use</button> */}
                
                <button type="submit" aria-label="go-to-payment-info">Go To Payment Info</button>
            </form>
        </div>
    )
}

// PropTypes
// ShippingInfoDisplay.propTypes = {
//     firstName: PropTypes.string.isRequired,
//     lastName: PropTypes.string.isRequired,
//     address1: PropTypes.string.isRequired,
//     address2: PropTypes.string,
//     city: PropTypes.string.isRequired,
//     stateProvince: PropTypes.string.isRequired,
//     country: PropTypes.string.isRequired
// }