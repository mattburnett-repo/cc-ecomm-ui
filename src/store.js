import { configureStore } from '@reduxjs/toolkit';

import authReducer from './features/auth/authSlice';
import productCategoryReducer from './features/productCategory/productCategorySlice'
import productsListingReducer from './features/products/productsSlice'
import ordersReducer from './features/orders/ordersSlice'
import cartsReducer from './features/carts/cartsSlice'

function rootReducer() {
    return {
        reducer: {
            auth: authReducer,
            productCategories: productCategoryReducer,
            orders: ordersReducer,
            carts: cartsReducer,
            products: productsListingReducer,
        }    
    }
}

const store = configureStore(rootReducer());

export default store;