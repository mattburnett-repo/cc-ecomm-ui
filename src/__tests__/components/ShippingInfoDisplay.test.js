// 20211217: Wireframe https://wireframe.cc/ag82hc

import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

import { Provider } from 'react-redux'
import store from '../../store'

import ShippingInfoDisplay from '../../components/shipping/ShippingInfoDisplay';

const mockSavedAddreses = []
const mockFunction = jest.fn()

function handlers() {
    return {
        handleSaveShippingInfo: mockFunction
    }
}

describe('<ShippingInfo /> component tests', () => {
    test.todo('Adding StyledComponents results in \'TypeError: Cannot read properties of undefined (reading \'light\')\' errors. Figure this out someday.')

    // beforeEach(() => render(<Provider store={store}><ShippingInfoDisplay savedAddresses={mockSavedAddreses} handlers={handlers()} /></Provider>));

    // it('should render', () => {
    //     screen.getByRole('presentation', {name: /shipping-info/i});
    // })
    // it('should render and display a first name', () => {
    //     const theElement = screen.getByRole('textbox', {name: /first-name/i});
    //     userEvent.type(theElement, 'first-name');
        
    //     expect(theElement).toHaveValue('first-name');
    // });
    // it('should render and display last name', () => {
    //     const theElement = screen.getByRole('textbox', {name: /last-name/i});
    //     userEvent.type(theElement, 'last-name');
        
    //     expect(theElement).toHaveValue('last-name');
    // });
    // it('should render and display address 1', () => {
    //     const theElement = screen.getByRole('textbox', {name: /address-1/i});
    //     userEvent.type(theElement, 'address-1');
        
    //     expect(theElement).toHaveValue('address-1');
    // });
    // it('should render and display address 2', () => {
    //     const theElement = screen.getByRole('textbox', {name: /address-2/i});
    //     userEvent.type(theElement, 'address-2');
        
    //     expect(theElement).toHaveValue('address-2');
    // });
    // it('should render and display city', () => {
    //     const theElement = screen.getByRole('textbox', {name: /city/i});
    //     userEvent.type(theElement, 'city');
        
    //     expect(theElement).toHaveValue('city');
    // });
    // it('should render and display state / province', () => {
    //     const theElement = screen.getByRole('textbox', {name: /state-province/i});
    //     userEvent.type(theElement, 'state-province');
        
    //     expect(theElement).toHaveValue('state-province');
    // });
    // it('should rendeer and display postal code', () => {
    //     const theElement = screen.getByRole('textbox', {name: /postal-code/i});
    //     userEvent.type(theElement, 'postal-code');
        
    //     expect(theElement).toHaveValue('postal-code');
    // });
    // it('should render and display country', () => {
    //     const theElement = screen.getByRole('textbox', {name: /country/i});
    //     userEvent.type(theElement, 'country');
        
    //     expect(theElement).toHaveValue('country');
    // });

    // it('should render and click a save-shipping-info button', () => {
    //     const theVal = screen.getByRole('button', {name: /save-shipping-info/i});
    //     fireEvent.submit(theVal);
    // });
    // it('should render and click a go-to-payment-info button', () => {
    //     const theVal = screen.getByRole('button', {name: /go-to-payment-info/i});
    //     fireEvent.submit(theVal);
    // });

    // it('creates a snapshot', () => {
    //     const tree = renderer.create(<Provider store={store}><ShippingInfoDisplay savedAddresses={mockSavedAddreses} handlers={handlers()} /></Provider>).toJSON();
    //     expect(tree).toMatchSnapshot();  
    // });
});

