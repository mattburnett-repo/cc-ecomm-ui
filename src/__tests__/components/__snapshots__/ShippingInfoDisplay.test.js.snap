// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShippingInfo /> component tests creates a snapshot 1`] = `
<div
  aria-label="shipping-info"
  role="presentation"
>
  <h3>
    Shipping Info
  </h3>
  <div
    aria-label="cart-summary-info"
    role="presentation"
  >
    Cart / Order has 
    0
     items, total price: FIXME
  </div>
  <form
    method="post"
  >
    <div>
      <label
        htmlFor="first-name"
      >
        First Name:
      </label>
      <input
        aria-label="first-name"
        id="first-name"
        name="firstName"
      />
    </div>
    <div>
      <label
        htmlFor="last-name"
      >
        Last Name:
      </label>
      <input
        aria-label="last-name"
        id="last-name"
        name="lastName"
      />
    </div>
    <div>
      <label
        htmlFor="address-1"
      >
        Address 1:
      </label>
      <input
        aria-label="address-1"
        id="address-1"
        name="address1"
      />
    </div>
    <div>
      <label
        htmlFor="address-2"
      >
        Address 2:
      </label>
      <input
        aria-label="address-2"
        id="address-2"
        name="address2"
      />
    </div>
    <div>
      <label
        htmlFor="city"
      >
        City:
      </label>
      <input
        aria-label="city"
        id="city"
        name="city"
      />
    </div>
    <div>
      <label
        htmlFor="state-province"
      >
        State / Province:
      </label>
      <input
        aria-label="state-province"
        id="state-province"
        name="stateProvince"
      />
    </div>
    <div>
      <label
        htmlFor="postal-code"
      >
        Postal Code:
      </label>
      <input
        aria-label="postal-code"
        id="postal-code"
        name="postalCode"
      />
    </div>
    <div>
      <label
        htmlFor="country"
      >
        Country:
      </label>
      <input
        aria-label="country"
        id="country"
        name="country"
      />
    </div>
    <button
      aria-label="go-to-payment-info"
      type="submit"
    >
      Go To Payment Info
    </button>
  </form>
</div>
`;
